modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro

```diff
@@ -0,0 +1,598 @@
+	macro createCaseResult2API {
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/c/caseresults \
+				-u test@liferay.com:test \
+                -H 'accept: application/json' \
+				-H 'Content-Type: application/json' \
+				-d '{
+					"dueStatus": {
+						"key": "${key}",
+						"name": "${statusCaseResult}"
+					},
+					"r_buildToCaseResult_c_buildId": "${buildId}",
+					"r_caseToCaseResult_c_caseId": "${casesId_2}",
+					"r_componentToCaseResult_c_componentId": "${componentId}",
+					"r_runToCaseResult_c_runId": "${runId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
+
+	@summary = "create third Case Result via API"
+	macro createCaseResult3API {
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/c/caseresults \
+				-u test@liferay.com:test \
+                -H 'accept: application/json' \
+				-H 'Content-Type: application/json' \
+				-d '{
+					"dueStatus": {
+						"key": "${key}",
+						"name": "${statusCaseResult}"
+					},
+					"r_buildToCaseResult_c_buildId": "${buildId}",
+					"r_caseToCaseResult_c_caseId": "${casesId_3}",
+					"r_componentToCaseResult_c_componentId": "${componentId}",
+					"r_runToCaseResult_c_runId": "${runId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
+
+	@summary = "create fourth Case Result via API"
+	macro createCaseResult4API {
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/c/caseresults \
+				-u test@liferay.com:test \
+                -H 'accept: application/json' \
+				-H 'Content-Type: application/json' \
+				-d '{
+					"dueStatus": {
+						"key": "${key}",
+						"name": "${statusCaseResult}"
+					},
+					"r_buildToCaseResult_c_buildId": "${buildId}",
+					"r_caseToCaseResult_c_caseId": "${casesId_4}",
+					"r_componentToCaseResult_c_componentId": "${componentId}",
+					"r_runToCaseResult_c_runId": "${runId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
+
+	@summary = "create a Case Result via API"
+	macro createCaseResultAPI {
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/c/caseresults \
+				-u test@liferay.com:test \
+                -H 'accept: application/json' \
+				-H 'Content-Type: application/json' \
+				-d '{
+					"dueStatus": {
+						"key": "${key}",
+						"name": "${statusCaseResult}"
+					},
+					"r_buildToCaseResult_c_buildId": "${buildId}",
+					"r_caseToCaseResult_c_caseId": "${casesId}",
+					"r_componentToCaseResult_c_componentId": "${componentId}",
+					"r_runToCaseResult_c_runId": "${runId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
```

Comment:
@joelrocha68, in this case, couldn't we just create 1 macro? The macro is doing the same thing, but just with different casesIDs